src/bit_str.jl

```diff
@@ -183,43 +166,42 @@ function parse_bit(::Type{T}, str::String) where {T <: Integer}
         else
             error("expect 0 or 1, got $each at $k-th bit")
         end
+        k<=bsizeof(T) || error("Bit String too long! use Long bit string?")
```

Comment:
this is wrong for `BigInt`, the type size of `BigInt` is just a pointer with an integer length, `BigInt` is actually an array for arbitrary integer precision, this will make `BigInt` only accept 16 digits...I think we need to separate the error msg with dispatch, e.g

```
# actual implementation
_parse_bit(::Type{T}, s)

# error msg
parse_bit(::Type{Int64}, s) = # error when it is larger than 64, and suggest to use lbit"" string literal
parse_bit(::Type{BigInt}, s) = # won't error at all
```

I think we should suggest to use the string literal in error msg since `parse_bit` is only used in string literal.

Reply:
I would prefer not changing it. This error message is more correct.

Reply:
The code itself it wrong. bsizeof(BigInt) doesn't mean anything.

Reply:
MWE:

```
BitBasis.parse_bit(BigInt, "10101010101010101010101010101010010101010101"^10)
ERROR: Bit String too long! use Long bit string?
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] parse_bit(::Type{BigInt}, ::String) at /Users/roger/.julia/dev/BitBasis/src/bit_str.jl:170
 [3] top-level scope at none:0
```

This doesn't make any sense.

Reply:
ok, I see.